File: src/lib/context.ts
import { loadProject } from './config.js';
import type { Project } from '../types.js';

/**
 * Build a context string to inject into Claude sessions spawned from Yeehaw.
 * Includes project name and wiki section titles (not content) to hint at available context.
 */
export function buildProjectContext(projectName: string): string | null {
  const project = loadProject(projectName);
  if (!project) {
    return null;
  }

  const lines: string[] = [];

  lines.push(`You are working on the "${project.name}" project.`);

  if (project.summary) {
    lines.push(`Project: ${project.summary}`);
  }

  // Add wiki section titles as hints
  const wikiSections = project.wiki || [];
  if (wikiSections.length > 0) {
    lines.push('');
    lines.push('Yeehaw wiki sections available:');
    for (const section of wikiSections) {
      lines.push(`- ${section.title}`);
    }
    lines.push('');
    lines.push('Use mcp__yeehaw__get_wiki_section to fetch relevant context before making architectural decisions.');
  }

  return lines.join('\n');
}

/**
 * Build context for a livestock-specific session.
 * Includes project context plus livestock details.
 */
export function buildLivestockContext(projectName: string, livestockName: string): string | null {
  const project = loadProject(projectName);
  if (!project) {
    return null;
  }

  const livestock = project.livestock?.find(l => l.name === livestockName);
  if (!livestock) {
    return buildProjectContext(projectName);
  }

  const lines: string[] = [];

  lines.push(`You are working on the "${project.name}" project, in the "${livestock.name}" environment.`);

  if (project.summary) {
    lines.push(`Project: ${project.summary}`);
  }

  // Add livestock details
  if (livestock.branch) {
    lines.push(`Branch: ${livestock.branch}`);
  }

  // Add wiki section titles as hints
  const wikiSections = project.wiki || [];
  if (wikiSections.length > 0) {
    lines.push('');
    lines.push('Yeehaw wiki sections available:');
    for (const section of wikiSections) {
      lines.push(`- ${section.title}`);
    }
    lines.push('');
    lines.push('Use mcp__yeehaw__get_wiki_section to fetch relevant context before making architectural decisions.');
  }

  return lines.join('\n');
}

File: src/context.rs
use crate::types::Project;

pub fn build_project_context(project: &Project) -> String {
    let mut lines = Vec::new();

    lines.push(format!("Project: {}", project.name));

    if let Some(ref summary) = project.summary {
        lines.push(format!("Summary: {}", summary));
    }

    if !project.wiki.is_empty() {
        lines.push(String::new());
        lines.push("Available wiki sections (use mcp__yeehaw__get_wiki_section to read):".to_string());
        for section in &project.wiki {
            lines.push(format!("  - {}", section.title));
        }
    }

    lines.join("\n")
}

pub fn build_livestock_context(project: &Project, livestock_name: &str) -> String {
    let mut context = build_project_context(project);

    if let Some(ls) = project.livestock.iter().find(|l| l.name == livestock_name) {
        context.push_str(&format!("\n\nLivestock: {}", ls.name));
        if let Some(ref branch) = ls.branch {
            context.push_str(&format!("\nBranch: {}", branch));
        }
        if let Some(ref repo) = ls.repo {
            context.push_str(&format!("\nRepo: {}", repo));
        }
    }

    context
}
